### DISCUSSION TITLE
CORS Policy issue

### DISCUSSION THREAD

**Discussion by charbs-io** [pockethost/pockethost#273] [Q&A] [answered]
`Access to fetch at 'https://appname.pockethost.io/api/collections/budgets/records?page=1&perPage=10&sort=created' from origin 'https://app.appname.com' has been blocked by CORS policy: Request header field workspace-id is not allowed by Access-Control-Allow-Headers in preflight response.`

Hi, I'm using PocketBase's `beforeSend` method from the JS SDK to insert a custom header with each request (`workspace-id`). This is specific to my app needs, and it enables a fairly complex multi tenant (sort of) scenario to happen.

PocketHost doesn't seem to like it though, is there anything I can do at all on the PocketHost side to fix this?

**Comment by benallfree** [comment]
Since you’re in the code already, take a look at the cors stuff in the ProxyService. Maybe you can send another PR :)

**Comment by charbs-io** [comment]
@benallfree - all I can think of is adding

`res.setHeader('Access-Control-Allow-Headers', '*')`

before we return the handler of the `middleware.push` method.

But I have no idea if this would present any security implications on production environments...

**Comment by charbs-io** [comment]
Here's what I could come up with...
https://github.com/charbs-io/pockethost/commit/1e538b199186cf391f45f4d50bf5622bc4cae56f

This certainly fixes the issue stated above (tested locally while running the PocketHost stack). But as I mentioned earlier, I have no idea if there are serious security implications from this.

**Reply by charbs-io** [reply]
@benallfree thoughts on the above?

**Comment by benallfree** [comment, marked as answer]
@charbs-io I've been thinking about it, and I think it would be okay to use your solution. The PocketHost code is really just a proxy, so I don't see any issue with allowing all headers. `pocketbase` can always deny downstream. Please submit a PR for me to review and I'll push it out asap :)

**Comment by charbs-io** [comment]
@benallfree I agree. I had to (temporarily) host my app on Fly.io and my setup had no issues. I've thought about this a bit more too and I think it's fine. Thanks for getting back to me. [PR Open](https://github.com/benallfree/pockethost/pull/274). I look forward to moving the hosting to PocketHost once this is done! (My app will also be a good production usecase for pb_hooks on PocketHost - a good way to test it.)

**Comment by leninriv** [comment]
Was this fixed? It still doesn't work for me

**Reply by benallfree** [reply]
Can you provide more detail please?

**Reply by leninriv** [reply]
thanks for answer.
sure.
i clone the repo in my computer, the master branch
i created the .env file 
i run the yarn install and  yarn dev
the server is runin g but im seeing this error in the console.

<img width="805" alt="Screenshot 2023-10-23 at 19 00 02" src="https://github.com/pockethost/pockethost/assets/25333291/84bd55a6-8549-4933-ada0-b53a833aec82">


i can access to "https://app.pockethost.lvh.me" from my browser. but im not able to login or create an account in my local.

im having this message from my browser.

<img width="1780" alt="Screenshot 2023-10-23 at 19 02 26" src="https://github.com/pockethost/pockethost/assets/25333291/665a4ce1-13b2-45fd-b97b-c8ed342d8b67">


thanks for your help!

**Reply by benallfree** [reply]
I see, I didn't realize you were trying to run the whole stack locally.

There is a lot of work being done in the next week or two. My best suggestion is to head over to our Discord server so we can keep in touch about updates and have you try some stuff.

**Reply by leninriv** [reply]
Cool. Thanks !
